### PR TITLE
Remove template fragment caching.

### LINKED ISSUE #8460: Remove template fragment caching
As part of #7743, splitted off to make review easier.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
* Removes template fragment-caching completely
* Adds a few performance optimizations, backported from #8435
* Removes the globally available `ADMIN_MESSAGE` and explicitly fetches
it in our base template. That way we don't call zadmin.get_config on
every template-render needlessly
* Improved developer-status querying, merged the caching from is_developer, is_addon_developer, and is_artist

Fixes #8460

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Indeed, I started tackling it but it turned out to be much more involved
than anticipated so I delayed it, especially since it didn't result in
any performance improvements on the bigger pages. I can back it out and
push an update once I land

On Mon, Jun 11, 2018, at 10:54 AM, Mathieu Pillard wrote:
> *@diox* commented on this pull request.


> 
> In src/olympia/addons/models.py[1]:


> > @@ -1009,11 +1009,18 @@ def attach_listed_authors(addons,
> > addon_dict=None): .filter(addons__in=addons, addonuser__listed=True)
> > .extra(select={'addon_id': 'addons_users.addon_id', 'position':
> > 'addons_users.position'})) -        qs = sorted(qs, key=lambda u:
> > (u.addon_id, u.position)) -        for addon_id, users in
> > itertools.groupby(qs, key=lambda u: u.addon_id): -
> > addon_dict[addon_id].listed_authors = list(users) -        # FIXME:
> > set listed_authors to empty list on addons without listed
>> This FIXME is still valid


> 
> In src/olympia/addons/models.py[2]:


> > @@ -1009,11 +1009,18 @@ def attach_listed_authors(addons,
> > addon_dict=None): .filter(addons__in=addons, addonuser__listed=True)
> > .extra(select={'addon_id': 'addons_users.addon_id', 'position':
> > 'addons_users.position'})) -        qs = sorted(qs, key=lambda u:
> > (u.addon_id, u.position)) -        for addon_id, users in
> > itertools.groupby(qs, key=lambda u: u.addon_id): -
> > addon_dict[addon_id].listed_authors = list(users) -        # FIXME:
> > set listed_authors to empty list on addons without listed
>> In fact, why change this at all? You're not dealing with add-ons with
> no authors, so I don't see what this change gives us.> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[3], or mute the thread[4].
Links:

  1. https://github.com/mozilla/addons-server/pull/8461#discussion_r194265693
  2. https://github.com/mozilla/addons-server/pull/8461#discussion_r194265720
  3. https://github.com/mozilla/addons-server/pull/8461#pullrequestreview-127405225

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1009,11 +1009,18 @@ def attach_listed_authors(addons, addon_dict=None):
               .filter(addons__in=addons, addonuser__listed=True)
               .extra(select={'addon_id': 'addons_users.addon_id',
                              'position': 'addons_users.position'}))
-        qs = sorted(qs, key=lambda u: (u.addon_id, u.position))
-        for addon_id, users in itertools.groupby(qs, key=lambda u: u.addon_id):
-            addon_dict[addon_id].listed_authors = list(users)
-        # FIXME: set listed_authors to empty list on addons without listed
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/addons/models.py]: reverted in e5f79fc5da

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -1009,11 +1009,11 @@ def attach_listed_authors(addons, addon_dict=None):
               .filter(addons__in=addons, addonuser__listed=True)
               .extra(select={'addon_id': 'addons_users.addon_id',
                              'position': 'addons_users.position'}))
+
         qs = sorted(qs, key=lambda u: (u.addon_id, u.position))
         for addon_id, users in itertools.groupby(qs, key=lambda u: u.addon_id):
             addon_dict[addon_id].listed_authors = list(users)
         # FIXME: set listed_authors to empty list on addons without listed
-        # authors.
```
Comment (diox) [inline src/olympia/addons/models.py]: keep this

File: src/olympia/addons/models.py [hunk 3]
```diff
@@ -1009,11 +1009,11 @@ def attach_listed_authors(addons, addon_dict=None):
               .filter(addons__in=addons, addonuser__listed=True)
               .extra(select={'addon_id': 'addons_users.addon_id',
                              'position': 'addons_users.position'}))
+
```
Comment (diox) [inline src/olympia/addons/models.py]: nit: maybe don't insert the newline as well right now

**Pull request merged**